src/main/java/uk/ac/cam/cl/dtg/segue/api/QuestionFacade.java

```diff
@@ -232,6 +261,69 @@ public Response getQuestionsAnswered(@Context final HttpServletRequest request,
     }
   }
 
+  /**
+   * REST end point to provide five questions in random located in the question tile of the student dashboard.
+   *
+   * @param request  - this allows us to check to see if a user is currently logged in.
+   * @param subjects - a comma separated list of subjects
+   * @return a Response containing a gameboard object or containing a SegueErrorResponse.
+   */
+  @GET
+  @Path("/randomQuestions")
+  @Produces(MediaType.APPLICATION_JSON)
+  @GZIP
+  public final Response getRandomQuestions(@Context final HttpServletRequest request,
+                                           @QueryParam("subjects") final String subjects) {
+    try {
+      RegisteredUserDTO currentUser = this.userManager.getCurrentRegisteredUser(request);
+      var filterQuestionsPreference = this.userPreferenceManager.getUserPreference(
+          "DISPLAY_SETTING",
+          "HIDE_NON_AUDIENCE_CONTENT",
+          currentUser.getId());
+
+      GameFilter gameFilter = new GameFilter();
+
+      if (filterQuestionsPreference != null && filterQuestionsPreference.getPreferenceValue()) {
```

Comment:
For better readability, this should be moved to a private method that returns a GameFilter (an empty one if this condition is false).